module/zfs/zfs_acl.c

```diff
@@ -53,6 +53,7 @@
 #include <sys/zap.h>
 #include <sys/sa.h>
 #include <sys/trace_acl.h>
+#include <linux/vfs_compat.h>
```

Comment:
Rather than including `linux/vfs_compat.h` I'd recommend including `sys/zpl.h`.  This will pull in additional headers *_compat.h headers which aren't needed, but it let's us further abstract away the Linux specific code.